hist/hist/src/TH1.cxx

```diff
@@ -2616,7 +2616,7 @@ TH1 *TH1::GetCumulative(Bool_t forward, const char* suffix) const
                hintegrated->AddBinContent(bin, sum);
                if (fSumw2.fN) {
                   esum += GetBinErrorSqUnchecked(bin);
-                  fSumw2.fArray[bin] = esum;
+                  hintegrated->GetSumw2()->SetAt(esum, bin);
```

Comment:
It is maybe shorter and more efficient to write: 
```
hintegrated->fSumw2.fArray[bin] = esum;
```

Reply:
Cool! But why are these protective attributes of a non-self TH1 instance accessible here?

Reply:
> It is maybe shorter and more efficient to write:
> 
> ```
> hintegrated->fSumw2.fArray[bin] = esum;
> ```

Updated and tested.